spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/mapping/SpannerPersistentEntityImpl.java

```diff
@@ -96,6 +97,8 @@
 
 	private final String where;
 
+	private final Map<Class<?>, String> jsonPropertiesClassToName = new HashMap<>();
```

Comment:
What is the purpose of this map again? Couldn't quite figure out based on the name.

Reply:
It's a map holding the class to field-name of Json annotated fields in the persistent entity. This gets produced once during the initializing process of persistent entity, so that in the added logic in `SqlSpannerQuery`, this information is accessible and does not need to loop over each persistent property to get if it's a json field. 

The logic added in `SqlSpannerQuery` is needed when the query returns a list or a single item of the Json field, not the whole entity or a simple field. This is an edge case that cannot be handled by the original logic, the code checks if the `returnedType` is one of the json field/s in the entity, if so, gets the corresponding field name from `jsonPropertiesClassToName` and proceed with `getSingleJsonValue` structAccessor function. 

Here is an [integration test](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/41be5213ed7e5bee7695f247604a2f8ae61b35a9/spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/it/SpannerRepositoryIntegrationTests.java#L531-L533) added for this edge case.